### PR TITLE
ARROW-7461: [Java] fix typos

### PR DESCRIPTION
**Pull request by kiszk**
This PR fixes typos in files under `java` directory.

### CODE REVIEW

**Review by kou — APPROVED** [review]
+1

File: java/flight/src/main/java/org/apache/arrow/flight/FlightServerMiddleware.java [hunk 1]
```diff
@@ -87,7 +87,7 @@
   void onCallCompleted(CallStatus status);
 
   /**
-   * Callback for when an RPC method implementation throws an uncaught exception.
+   * Callback for when an RPC method implementation throws a uncaught exception.
```
Comment (liyafan82) [inline java/flight/src/main/java/org/apache/arrow/flight/FlightServerMiddleware.java]: This is a litle nit:
According to my (poor) English knowledge, it should be "an" here? Could someone please help explain it? :-)

Comment (kou) [reply java/flight/src/main/java/org/apache/arrow/flight/FlightServerMiddleware.java]: Oh, sorry. I missed this.
@kiszk Could you open a follow-up pull request?

Comment (kiszk) [reply java/flight/src/main/java/org/apache/arrow/flight/FlightServerMiddleware.java]: Oh, sorry this is my mistake... Should be "an".
I will fix this in another PR (probably #6089).

Comment (kiszk) [reply java/flight/src/main/java/org/apache/arrow/flight/FlightServerMiddleware.java]: @liyafan82 @kou I am sorry for my bad. I created a follow-up PR #6090

Comment (liyafan82) [reply java/flight/src/main/java/org/apache/arrow/flight/FlightServerMiddleware.java]: @kiszk and @kou Thanks a lot for your effort.

**Pull request closed without merging**